File: apps/web/types/project.ts
export interface Project {
  id: string;
  workspace_id: string;
  title: string;
  created_at: string;
  updated_at: string;
}

export interface CreateProjectRequest {
  title: string;
}

export interface UpdateProjectRequest {
  title?: string;
}

export interface ProjectListResponse {
  projects: Project[];
}

export interface ProjectWithWorkspace extends Project {
  workspace?: {
    id: string;
    name: string;
  };
}

File: apps/api/schemas/projectSchemas/project.go
package projectSchemas

import "time"

type Project struct {
	ID          string    `json:"id"`
	WorkspaceID string    `json:"workspace_id"`
	Title       string    `json:"title"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type CreateProjectRequest struct {
	Title string `json:"title" binding:"required"`
}
